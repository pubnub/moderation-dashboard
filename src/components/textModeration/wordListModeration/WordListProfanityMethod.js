import React from "react";
import {
  Grid,
  Button,
  Box,
  Card,
  Chip,
  Typography,
  TextField,
  Select,
  Checkbox,
  CircularProgress,
  FormControl,
  MenuItem,
} from "@material-ui/core";
import { getProfanityWordsByLanguage, constantBoolean } from "../../../utils/helpers";
import { useStyles } from "../../../style/textModeration.js";
import { handleChange } from "./WordListHandler";
import { LightTooltip } from "../../../style/tooltip";

const WordListProfanityMethod = ({
  handleSave,
  defaultWords,
  state,
  setState,
  profanityList,
  setProfanityList,
}) => {
  const classes = useStyles();
  const languageList = ["English", "Hindi", "Spanish", "Portugese", "French"];

  const {
    wordListChannel,
    wordListLanguage,
    wordListModType,
    wordsListPatternError,
    wordListReRouteMessages,
    wordsListMaskCharError,
    wordsListChannelError,
    applyToAllChannelIdsWordlist,
    wordListCharacterToMaskWith,
  } = state.wordList;

  const checkboxForReroute = constantBoolean(wordListReRouteMessages);
  const checkForApplyToAllChannelIds = constantBoolean(applyToAllChannelIdsWordlist);

  return (
    <>
      <Card className={classes.profanityCardRoot}>
        <Grid container>
          <Grid item>
            <label testId="Channel_Id" className={classes.labelStyle}>
              Channel ID
            </label>
          </Grid>
          <Grid item>
            <Box className={classes.infoIcon}>
              <LightTooltip title="Note: Please add a Channel ID or a Channel pattern. eg. channel.* OR pubNub. Channel ID pattern applies to BOTH text & image moderation if both are enabled.">
                <img src={process.env.PUBLIC_URL + "/images/info-circle.svg"} alt="info-circle" />
              </LightTooltip>
            </Box>
          </Grid>
        </Grid>
        <Box pt={1} mb={2}>
          <TextField
            id="channelId"
            placeholder="Channel"
            name="wordListChannel"
            disabled={checkForApplyToAllChannelIds}
            error={wordsListChannelError}
            variant="outlined"
            size="small"
            fullWidth
            value={wordListChannel}
            onChange={handleChange({
              setState,
              state,
              profanityList,
              setProfanityList,
            })}
          />
        </Box>
        <Grid item>
          <Grid container justify="flex-start">
            <Grid item>
              <Checkbox
                id="allChannelIdcheckBox"
                checked={checkForApplyToAllChannelIds}
                onChange={handleChange({
                  setState,
                  state,
                  name: "applyToAllChannelIdsWordlist",
                })}
              />
            </Grid>
            <Grid item>
              <Box pt={1}>
                <Typography testid="allChannelIdText" variant="body1">
                  Apply to All Channel IDs
                </Typography>
              </Box>
            </Grid>
          </Grid>
        </Grid>
        <br />
        <Grid container justify="flex-start">
          <Grid item sm={5}>
            <label testId="language" className={classes.labelStyle}>
              Language
            </label>
            <Box pt={1} mb={2}>
              <FormControl
                className={classes.formControl}
                size="small"
                placeholder="Langauge"
                fullWidth
              >
                <Select
                  id="wordListLanguage"
                  variant="outlined"
                  MenuProps={{
                    anchorOrigin: {
                      vertical: "bottom",
                      horizontal: "left",
                    },
                    getContentAnchorEl: null,
                  }}
                  name="wordListLanguage"
                  value={wordListLanguage}
                  onChange={handleChange({
                    setState,
                    state,
                    profanityList,
                    setProfanityList,
                  })}
                  inputProps={{ "aria-label": "Without label" }}
                >
                  {languageList.map((language, index) => {
                    return (
                      <MenuItem value={language} key={index}>
                        {language}
                      </MenuItem>
                    );
                  })}
                </Select>
              </FormControl>
            </Box>
          </Grid>
        </Grid>
        <br />
        <Grid container justify="space-between">
          <Grid item>
            <Grid container>
              <Grid item>
                <label testId="WordList" className={classes.labelStyle}>
                  Word List
                </label>
              </Grid>
              <Grid item>
                <Box className={classes.infoIcon}>
                  <LightTooltip title="Note: Please add comma after a every word. No spaces are allowed">
                    <img
                      src={process.env.PUBLIC_URL + "/images/info-circle.svg"}
                      alt="info-circle"
                    />
                  </LightTooltip>
                </Box>
              </Grid>
            </Grid>
          </Grid>
          <Grid item>
            <Typography
              testid="defaultWordsLabel"
              align="right"
              className={classes.defaultText}
              onClick={() => defaultWords(wordListLanguage)}
            >
              Use Default words
            </Typography>
          </Grid>
        </Grid>
        <Box pt={1} mb={2}>
          <TextField
            id="defaultWordsBox"
            placeholder="comma,separated,list,of,words"
            className={classes.languageWords}
            multiline
            error={wordsListPatternError}
            rowsMax={6}
            name="wordListLanguageWords"
            value={getProfanityWordsByLanguage(profanityList, wordListLanguage)}
            onChange={handleChange({
              setState,
              state,
              profanityList,
              setProfanityList,
            })}
            fullWidth
            variant="outlined"
          />
        </Box>
        <br />
        <br />
        <Grid container justify="flex-start" spacing={6}>
          <Grid item sm={6} md={6}>
            <label testId="MaskOrBlock" className={classes.labelStyle}>
              When Profanity is detected
            </label>
            <Box pt={1} mb={2}>
              <FormControl className={classes.formControl} size="small" fullWidth>
                <Select
                  id="MaskOrBlockOption"
                  variant="outlined"
                  MenuProps={{
                    anchorOrigin: {
                      vertical: "bottom",
                      horizontal: "left",
                    },
                    getContentAnchorEl: null,
                  }}
                  name="wordListModtype"
                  onChange={handleChange({
                    setState,
                    state,
                    profanityList,
                    setProfanityList,
                  })}
                  value={wordListModType}
                  inputProps={{ "aria-label": "Without label" }}
                >
                  <MenuItem value="Mask-word">Mask Word</MenuItem>
                  <MenuItem value="Block-message">Block Message</MenuItem>
                </Select>
              </FormControl>
            </Box>
          </Grid>
          <Grid item sm={6} md={6}>
            {wordListModType === "Mask-word" && (
              <>
                <Grid container>
                  <Grid item>
                    <label testId="maskLabel" className={classes.label}>
                      Set a masking character
                    </label>
                  </Grid>
                  <Grid item>
                    <Box className={classes.infoIcon}>
                      <LightTooltip title="Note: Default value will be *">
                        <img
                          src={process.env.PUBLIC_URL + "/images/info-circle.svg"}
                          alt="info-circle"
                        />
                      </LightTooltip>
                    </Box>
                  </Grid>
                </Grid>

                <Box pt={1}>
                  <TextField
                    id="maskChar"
                    size="small"
                    placeholder="Set"
                    inputProps={{ maxLength: 1 }}
                    error={wordsListMaskCharError}
                    className={classes.languageWords}
                    value={wordListCharacterToMaskWith}
                    onChange={handleChange({
                      setState,
                      state,
                      profanityList,
                      setProfanityList,
                    })}
                    fullWidth
                    name="wordListCharacterToMaskWith"
                    variant="outlined"
                  />
                </Box>
              </>
            )}
          </Grid>
        </Grid>
        <br />
        <Grid container justify="flex-start">
          <Grid item>
            <Grid container>
              <Grid item>
                <Checkbox
                  id="reRouteCheckBox"
                  name="wordListRerouteMessage"
                  onChange={handleChange({
                    setState,
                    state,
                    profanityList,
                    setProfanityList,
                  })}
                  checked={checkboxForReroute}
                />
              </Grid>
              <Grid item>
                <Box pt={1}>
                  <Typography testid="ReRouteLabel" variant="body1">
                    Route messages to
                  </Typography>
                </Box>
              </Grid>
            </Grid>
          </Grid>
          <Grid item>
            <Box>
              <Chip className={classes.reroute} label={"banned." + wordListChannel} />
            </Box>
          </Grid>
        </Grid>

        <br />
        <Grid container justify="space-between" spacing={2}>
          <Grid item></Grid>
          <Grid item>
            <Button
              id="save"
              disabled={state.saveLoading}
              className={classes.saveButton}
              onClick={handleSave}
              variant="contained"
            >
              {state.saveLoading ? (
                <CircularProgress
                  size={25}
                  thickness={4}
                  color="primary"
                  className={classes.loader}
                />
              ) : (
                <></>
              )}
              Save
            </Button>
          </Grid>
        </Grid>
      </Card>
    </>
  );
};

export default WordListProfanityMethod;
