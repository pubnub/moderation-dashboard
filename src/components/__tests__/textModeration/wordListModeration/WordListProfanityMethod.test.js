import React from "react";
import { shallow } from "enzyme";
import WordListProfanityMethod from "../../../textModeration/wordListModeration/WordListProfanityMethod";

const mockProps = {
  wordList: {
    wordListChannel: "initialText",
    applyToAllChannelIdsWordlist: true,
    wordsListPatternError: false,
    wordListLanguage: "Hindi",
    wordListModType: "Mask-word",
    wordListReRouteMessages: false,
    wordsListMaskCharError: false,
    wordsListChannelError: false,
    wordListCharacterToMaskWith: "*",
  },
  textModerationToggle: false,
  wordListProfanity: false,
  automaticProfanity: false,
  channelOnChange: false,
  saveLoading: false,
  initialLoading: true,
  errorStatus: false,
  successStatus: false,
  errorMsg: "",
  successMsg: "",
};
const mockProfanityList = {
  English: "",
  Hindi: "Bakwaas, Chutiye",
  French: "",
  Portugese: "",
  Spanish: "",
};

describe("Word List Text Moderation", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <WordListProfanityMethod
        state={mockProps}
        profanityList={mockProfanityList}
        setState={jest.fn()}
        setProfanityList={jest.fn()}
      />
    );
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Contains Labels", () => {
    expect(wrapper.find("[testId='Channel_Id']").text()).toBe("Channel ID");
    expect(wrapper.find('[testid="allChannelIdText"]').text()).toBe("Apply to All Channel IDs");
    expect(wrapper.find('[testId="language"]').text()).toBe("Language");
    expect(wrapper.find('[testId="WordList"]').text()).toBe("Word List");
    expect(wrapper.find('[testid="defaultWordsLabel"]').text()).toBe("Use Default words");
    expect(wrapper.find('[testId="MaskOrBlock"]').text()).toBe("When Profanity is detected");
    expect(wrapper.find('[testId="maskLabel"]').text()).toBe("Set a masking character");
    expect(wrapper.find('[testid="ReRouteLabel"]').text()).toBe("Route messages to");
  });

  test("Check initial Values", () => {
    expect(wrapper.find("#channelId").getElement().props.value).toBe("initialText");
    expect(wrapper.find("#wordListLanguage").getElement().props.value).toBe("Hindi");
    expect(wrapper.find("#defaultWordsBox").getElement().props.value).toBe("Bakwaas, Chutiye");
    expect(wrapper.find("#MaskOrBlockOption").getElement().props.value).toBe("Mask-word");
    expect(wrapper.find("#maskChar").getElement().props.value).toBe("*");

    expect(wrapper.find("#channelId").getElement().props.placeholder).toBe("Channel");
    expect(wrapper.find("#defaultWordsBox").getElement().props.placeholder).toBe(
      "comma,separated,list,of,words"
    );

    expect(wrapper.find("#maskChar").getElement().props.placeholder).toBe("Set");
  });

  test("Check CheckBoxes", () => {
    expect(wrapper.find("#allChannelIdcheckBox").getElement().props.checked).toBe(true);
    expect(wrapper.find("#reRouteCheckBox").getElement().props.checked).toBe(false);
  });

  test("Word Handler", () => {
    const ChannelChange = wrapper.find("#channelId").simulate("change", {
      target: { name: "wordListChannel", value: "initialText" },
    });
    expect(ChannelChange).toBeTruthy();

    const LangChange = wrapper.find("#wordListLanguage").simulate("change", {
      target: { name: "wordListLanguage", value: "Hindi" },
    });
    expect(LangChange).toBeTruthy();

    const WordChange = wrapper.find("#defaultWordsBox").simulate("change", {
      target: { name: "wordListLanguageWords", value: "Hindi" },
    });
    expect(WordChange).toBeTruthy();

    const WordListChange = wrapper.find("#MaskOrBlockOption").simulate("change", {
      target: { name: "wordListModtype", value: "Mask-word" },
    });
    expect(WordListChange).toBeTruthy();

    const MaskCharChange = wrapper.find("#maskChar").simulate("change", {
      target: { name: "wordListCharacterToMaskWith", value: "*" },
    });
    expect(MaskCharChange).toBeTruthy();

    const ReRouteChange = wrapper.find("#reRouteCheckBox").simulate("change", {
      target: { name: "wordListRerouteMessage", checked: true },
    });
    expect(ReRouteChange).toBeTruthy();

    const AllChannelTrue = wrapper.find("#allChannelIdcheckBox").simulate("change", {
      target: { checked: true },
    });
    expect(AllChannelTrue).toBeTruthy();

    const AllChannelFalse = wrapper.find("#allChannelIdcheckBox").simulate("change", {
      target: { checked: false },
    });
    expect(AllChannelFalse).toBeTruthy();
  });

  test("Check Save Button", () => {
    expect(wrapper.find("#save").text()).toBe("Save");
  });
});
