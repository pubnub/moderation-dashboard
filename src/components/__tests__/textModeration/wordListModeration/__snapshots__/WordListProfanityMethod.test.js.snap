// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Word List Text Moderation Snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-profanityCardRoot-11"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <label
          className="makeStyles-labelStyle-8"
          testId="Channel_Id"
        >
          Channel ID
        </label>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Styled(MuiBox)
          className="makeStyles-infoIcon-20"
        >
          <WithStyles(WithStyles(ForwardRef(Tooltip)))
            title="Note: Please add a Channel ID or a Channel pattern. eg. channel.* OR pubNub. Channel ID pattern applies to BOTH text & image moderation if both are enabled."
          >
            <img
              alt="info-circle"
              src="/images/info-circle.svg"
            />
          </WithStyles(WithStyles(ForwardRef(Tooltip)))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <Styled(MuiBox)
      mb={2}
      pt={1}
    >
      <WithStyles(ForwardRef(TextField))
        disabled={true}
        error={false}
        fullWidth={true}
        id="channelId"
        name="wordListChannel"
        onChange={[Function]}
        placeholder="Channel"
        size="small"
        value="initialText"
        variant="outlined"
      />
    </Styled(MuiBox)>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="flex-start"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Checkbox))
            checked={true}
            id="allChannelIdcheckBox"
            onChange={[Function]}
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <Styled(MuiBox)
            pt={1}
          >
            <WithStyles(ForwardRef(Typography))
              testid="allChannelIdText"
              variant="body1"
            >
              Apply to All Channel IDs
            </WithStyles(ForwardRef(Typography))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        sm={5}
      >
        <label
          className="makeStyles-labelStyle-8"
          testId="language"
        >
          Language
        </label>
        <Styled(MuiBox)
          mb={2}
          pt={1}
        >
          <WithStyles(ForwardRef(FormControl))
            fullWidth={true}
            placeholder="Langauge"
            size="small"
          >
            <WithStyles(ForwardRef(Select))
              MenuProps={
                Object {
                  "anchorOrigin": Object {
                    "horizontal": "left",
                    "vertical": "bottom",
                  },
                  "getContentAnchorEl": null,
                }
              }
              id="wordListLanguage"
              inputProps={
                Object {
                  "aria-label": "Without label",
                }
              }
              name="wordListLanguage"
              onChange={[Function]}
              value="Hindi"
              variant="outlined"
            >
              <WithStyles(ForwardRef(MenuItem))
                key="0"
                value="English"
              >
                English
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                key="1"
                value="Hindi"
              >
                Hindi
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                key="2"
                value="Spanish"
              >
                Spanish
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                key="3"
                value="Portugese"
              >
                Portugese
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                key="4"
                value="French"
              >
                French
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Select))>
          </WithStyles(ForwardRef(FormControl))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="space-between"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <label
              className="makeStyles-labelStyle-8"
              testId="WordList"
            >
              Word List
            </label>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <Styled(MuiBox)
              className="makeStyles-infoIcon-20"
            >
              <WithStyles(WithStyles(ForwardRef(Tooltip)))
                title="Note: Please add comma after a every word. No spaces are allowed"
              >
                <img
                  alt="info-circle"
                  src="/images/info-circle.svg"
                />
              </WithStyles(WithStyles(ForwardRef(Tooltip)))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          align="right"
          className="makeStyles-defaultText-16"
          onClick={[Function]}
          testid="defaultWordsLabel"
        >
          Use Default words
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <Styled(MuiBox)
      mb={2}
      pt={1}
    >
      <WithStyles(ForwardRef(TextField))
        error={false}
        fullWidth={true}
        id="defaultWordsBox"
        multiline={true}
        name="wordListLanguageWords"
        onChange={[Function]}
        placeholder="comma,separated,list,of,words"
        rowsMax={6}
        value="Bakwaas, Chutiye"
        variant="outlined"
      />
    </Styled(MuiBox)>
    <br />
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="flex-start"
      spacing={6}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={6}
        sm={6}
      >
        <label
          className="makeStyles-labelStyle-8"
          testId="MaskOrBlock"
        >
          When Profanity is detected
        </label>
        <Styled(MuiBox)
          mb={2}
          pt={1}
        >
          <WithStyles(ForwardRef(FormControl))
            fullWidth={true}
            size="small"
          >
            <WithStyles(ForwardRef(Select))
              MenuProps={
                Object {
                  "anchorOrigin": Object {
                    "horizontal": "left",
                    "vertical": "bottom",
                  },
                  "getContentAnchorEl": null,
                }
              }
              id="MaskOrBlockOption"
              inputProps={
                Object {
                  "aria-label": "Without label",
                }
              }
              name="wordListModtype"
              onChange={[Function]}
              value="Mask-word"
              variant="outlined"
            >
              <WithStyles(ForwardRef(MenuItem))
                value="Mask-word"
              >
                Mask Word
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                value="Block-message"
              >
                Block Message
              </WithStyles(ForwardRef(MenuItem))>
            </WithStyles(ForwardRef(Select))>
          </WithStyles(ForwardRef(FormControl))>
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        md={6}
        sm={6}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <label
              testId="maskLabel"
            >
              Set a masking character
            </label>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <Styled(MuiBox)
              className="makeStyles-infoIcon-20"
            >
              <WithStyles(WithStyles(ForwardRef(Tooltip)))
                title="Note: Default value will be *"
              >
                <img
                  alt="info-circle"
                  src="/images/info-circle.svg"
                />
              </WithStyles(WithStyles(ForwardRef(Tooltip)))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
        <Styled(MuiBox)
          pt={1}
        >
          <WithStyles(ForwardRef(TextField))
            error={false}
            fullWidth={true}
            id="maskChar"
            inputProps={
              Object {
                "maxLength": 1,
              }
            }
            name="wordListCharacterToMaskWith"
            onChange={[Function]}
            placeholder="Set"
            size="small"
            value="*"
            variant="outlined"
          />
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Checkbox))
              checked={false}
              id="reRouteCheckBox"
              name="wordListRerouteMessage"
              onChange={[Function]}
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <Styled(MuiBox)
              pt={1}
            >
              <WithStyles(ForwardRef(Typography))
                testid="ReRouteLabel"
                variant="body1"
              >
                Route messages to
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Styled(MuiBox)>
          <WithStyles(ForwardRef(Chip))
            className="makeStyles-reroute-18"
            label="banned.initialText"
          />
        </Styled(MuiBox)>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <br />
    <WithStyles(ForwardRef(Grid))
      container={true}
      justify="space-between"
      spacing={2}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      />
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-saveButton-13"
          disabled={false}
          id="save"
          variant="contained"
        >
          Save
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Card))>
</Fragment>
`;
